### PR TITLE
Added more tests for config validation

### PR DESCRIPTION
**Pull request by agnivade**

### CODE REVIEW

**Comment by agnivade** [comment]
@eapache  - 
```
Error while executing topic command : replication factor: 3 larger than available brokers: 2
[2017-10-06 10:11:17,820] ERROR org.apache.kafka.common.errors.InvalidReplicationFactorException: replication factor: 3 larger than available brokers: 2
```
Does not look like related to my change.

File: config_test.go [hunk 1]
```diff
@@ -33,6 +33,205 @@ func TestEmptyClientIDConfigValidates(t *testing.T) {
 	}
 }
 
+func TestNetConfigValidates(t *testing.T) {
+	tests := []struct {
+		name string
+		cfg  func() *Config // resorting to using a function as a param because of internal composite structs
```
Comment (eapache) [inline config_test.go]: change the method sig to `func(*Config)` and have the loop create the new config each time; will reduce the size of all the function bodies

**Comment by eapache** [comment]
Sometimes that CI error happens if the kafka cluster does not boot properly, I can trigger a re-run if it happens again.

File: config_test.go (hunk 1 above)
Comment (agnivade) [reply config_test.go]: Thanks. Updated.

**Comment by eapache** [comment]
Thanks!

**Pull request merged**